### PR TITLE
Implemented graphql gateway update interval

### PR DESCRIPTION
**Pull request by hansehe**
The service works well, but I have a problem when backend services are updated and the graphql gateway doesn't fetch the new changes. This solution should simply solve the problem.

### CODE REVIEW

**Comment by jakubknejzlik** [comment]
Hello @hansehe,  thanks for Your contribution. We are deploying this gateway usually in k8s or lambda and using following approach:
- k8s: run `kubectl rollout restart deployment/my-fancy-gateway`
- lambda: updating the description of lambda 😄 

...so this approach is really welcome. Also there's a plan to create schema repository where You can push the schema from each service. Right now if one of the service is down during the initialization, it is basically removed from schema.

File: index.js [hunk 1]
```diff
@@ -4,18 +4,30 @@ const { getApolloServer } = require("./apollo");
...
   const server = await getApolloServer();
   server.applyMiddleware({ app });
 
+  const activateUpdateGatewayInterval = getENV("GRAPHQL_UPDATE_GATEWAY", "false") === "true";
+  const updateGatewayInterval = getENV("GRAPHQL_UPDATE_GATEWAY_INTERVAL_MS", "60000");
+  if (activateUpdateGatewayInterval === true)
+  {
+    setInterval(async () => {
+      const server = await getApolloServer();
+      server.applyMiddleware({ app });
```
Comment (jakubknejzlik) [inline index.js:19]: 2nd part of update
```suggestion
      middleware = server.getMiddleware({});
```

File: index.js [hunk 2]
```diff
@@ -4,18 +4,30 @@ const { getApolloServer } = require("./apollo");
 
 (async () => {
   const app = express();
-  app.use(express.json({ limit: "2mb" }));
+  const jsonBodyLimit = getENV("GRAPHQL_JSON_BODY_LIMIT", "2mb");
+  app.use(express.json({ limit: jsonBodyLimit }));
 
   const server = await getApolloServer();
   server.applyMiddleware({ app });
```
Comment (jakubknejzlik) [inline index.js:11]: I'm not sure if we want to apply middleware each time it's probably pushing new middleware to stack and not removing the previous one. This basically creates dynamic middleware. The same would apply to lambda handler as well. Any thoughs about this?

1st part of update
```suggestion
    let middleware = server.getMiddleware({});

  app.use((req, res, next) => {
    middleware(req, res, next);
  });
```

Comment (hansehe) [reply index.js]: I tried your solution, and it works better I think. So I implemented that instead. The schema repository idea sounds good too! I would love to get noticed when it's ready to test out

File: index.js [hunk 3]
```diff
@@ -4,18 +4,41 @@ const { getApolloServer } = require("./apollo");
...
+  if (activateUpdateGatewayInterval === true)
+  {
+    setInterval(async () => {
+      // console.log("Updating graphql gateway..");
+      try {
+        const server = await getApolloServer();
+        let middleware = server.getMiddleware({});
+        app.use((req, res, next) => {
+          middleware(req, res, next);
+        });
```
Comment (jakubknejzlik) [inline index.js:27]:
```suggestion
        middleware = server.getMiddleware({});
```

Comment (jakubknejzlik) [reply index.js:27]: Is there a need to use `app.use`? Shouldn't be updating the middleware instance be enough?

Comment (hansehe) [reply index.js:27]: No, that's correct. I'll change it

**Comment by jakubknejzlik** [comment]
@hansehe I'm checking the latest commit, but we probably not understood each other :) 

What I meant by updating the middleware instance is:
```
const { getENV } = require("./env");
const express = require("express");
const { getApolloServer } = require("./apollo");

(async () => {
  const app = express();
  const jsonBodyLimit = getENV("GRAPHQL_JSON_BODY_LIMIT", "2mb");
  app.use(express.json({ limit: jsonBodyLimit }));

  const server = await getApolloServer();
  server.applyMiddleware({ app });

  const activateUpdateGatewayInterval = getENV("GRAPHQL_UPDATE_GATEWAY", "false") === "true";
  const updateGatewayInterval = getENV("GRAPHQL_UPDATE_GATEWAY_INTERVAL_MS", "60000");
  if (activateUpdateGatewayInterval === true)
  {
    setInterval(async () => {
      try {
        const server = await getApolloServer();
        middleware = server.getMiddleware({}); // just assign new instance
      } catch (error) {
        console.error(error);
      }
    }, updateGatewayInterval);
  }

  const HOST = getENV("HOST", "http://localhost");
  const PORT = getENV("PORT", "80");

  app.listen({ port: PORT }, () => {
    console.log(`🚀 Server ready at ${HOST}:${PORT}/graphql`);
  });
})();

module.exports.getServer = getApolloServer;
```

**Comment by hansehe** [comment]
Ok, thanks again! I updated the code with your reviews and it works with my local testing.

**Comment by jakubknejzlik** [comment]
Perfect, lookgs ok! Thanks for Your patience :)

**Comment by jakubknejzlik** [comment]
Hello @hansehe again, sorry for bringing this up, but I just noticed that there's `experimental_pollInterval`. Although there's this thread: https://github.com/apollographql/apollo-server/discussions/4280 I believe switching to it could be easier. 
What do You think?

**Pull request merged**